### PR TITLE
GDPR-78 Adding a variable to determine if the user should be able to …

### PR DESCRIPTION
**Pull request by brightonsbox**
…add details to a retention reason

### CODE REVIEW

File: src/main/resources/db/migration/V2_10__retention_reason_code_update.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE RETENTION_REASON_CODE ADD ALLOW_REASON_DETAILS BOOLEAN DEFAULT FALSE NOT NULL;
+COMMENT ON COLUMN RETENTION_REASON_CODE.ALLOW_REASON_DETAILS  IS 'Indication of whether the end user is allowed to comment on the reason they have provided';
+
+UPDATE RETENTION_REASON_CODE SET ALLOW_REASON_DETAILS = TRUE WHERE RETENTION_REASON_CODE_ID = 'OTHER';
+UPDATE RETENTION_REASON_CODE SET DISPLAY_NAME = 'MAPPA (Multi-Agency Public Protection Agreement)' WHERE RETENTION_REASON_CODE_ID = 'MAPPA';
```
Comment (brightonsbox) [inline src/main/resources/db/migration/V2_10__retention_reason_code_update.sql]: Also taking the opportunity to correct the display name for MAPPA

File: src/main/resources/application-localstack.yml [hunk 2]
```diff
@@ -18,3 +18,5 @@ sns:
   topic.arn: arn:aws:sns:eu-west-2:000000000000:offender_events
   endpoint.url: http://localstack:4575
   region: eu-west-2
+
+spring.security.oauth2.resourceserver.jwt.public-key-location: classpath:local-public-key.pub
```
Comment (brightonsbox) [inline src/main/resources/application-localstack.yml]: Unrelated change to make running locally work correctly

File: src/main/java/uk/gov/justice/hmpps/datacompliance/web/dto/ManualRetentionReasonDisplayName.java [hunk 3]
```diff
@@ -27,6 +27,17 @@
     private ManualRetentionReasonCode reasonCode;
 
     @ApiModelProperty(value = "The UI display name")
+    @NotNull
```
Comment (matt-crowson-moj) [inline src/main/java/uk/gov/justice/hmpps/datacompliance/web/dto/ManualRetentionReasonDisplayName.java]: Just mentioning in case there are other DTO's as well as this one.  No examples or position in swagger docs/annotations, meh.

Comment (brightonsbox) [reply src/main/java/uk/gov/justice/hmpps/datacompliance/web/dto/ManualRetentionReasonDisplayName.java]: I'll look to flesh it out, good point
